src/ui/preferences.ui

```diff
@@ -25,289 +25,161 @@
     <property name="step-increment">1</property>
     <property name="page-increment">10</property>
   </object>
-  <template class="CurtailPrefsWindow" parent="GtkWindow">
+  <template class="CurtailPrefsWindow" parent="AdwPreferencesWindow">
     <property name="title" translatable="yes">Preferences</property>
-    <property name="resizable">false</property>
-    <property name="child">
-      <object class="GtkStack" id="stack">
-        <property name="margin-start">20</property>
-        <property name="margin-end">20</property>
-        <property name="margin-top">20</property>
-        <property name="margin-bottom">20</property>
+    <property name="modal">True</property>
+    <property name="hide-on-close">True</property>
+    <child>
+      <object class="AdwPreferencesPage">
+        <property name="name">general</property>
+        <property name="title" translatable="yes">General</property>
+        <property name="icon-name">applications-system-symbolic</property>
         <child>
-          <object class="GtkStackPage">
-            <property name="name">general</property>
-            <property name="title" translatable="yes">General</property>
-            <property name="child">
-              <object class="GtkGrid" id="grid1">
-                <property name="valign">start</property>
-                <property name="margin-top">20</property>
-                <property name="margin-bottom">20</property>
-                <property name="row-spacing">20</property>
-                <property name="column-spacing">20</property>
+          <object class="AdwPreferencesGroup">
+            <child>
+              <object class="AdwActionRow">
+                <property name="title" translatable="yes">Save the compressed image into a new file</property>
+                <property name="activatable-widget">toggle_new_file</property>
                 <child>
-                  <object class="GtkLabel" id="new_file_label">
-                    <property name="halign">start</property>
+                  <object class="GtkSwitch" id="toggle_new_file">
                     <property name="valign">center</property>
-                    <property name="label" translatable="yes">Suffix to append at end of new file</property>
-                    <layout>
-                      <property name="column">0</property>
-                      <property name="row">3</property>
-                    </layout>
                   </object>
                 </child>
+              </object>
+            </child>
+            <child>
+              <object class="AdwActionRow">
```

Comment:
This might be a usecase for AdwEntryRow.

Reply:
Much better, thanks!